components/lcd/esp_lcd_ili9881c/esp_lcd_ili9881c.c

```diff
@@ -302,151 +110,436 @@ esp_err_t esp_lcd_new_panel_ili9881c(const esp_lcd_panel_io_handle_t io, const e
+    {0xC4, (uint8_t []){0x14}, 1, 0},
+    {0xC5, (uint8_t []){0x28}, 1, 0},
+    {0xC6, (uint8_t []){0x1D}, 1, 0},
+    {0xC7, (uint8_t []){0x21}, 1, 0},
+    {0xC8, (uint8_t []){0x6C}, 1, 0},
+    {0xC9, (uint8_t []){0x1B}, 1, 0},
+    {0xCA, (uint8_t []){0x26}, 1, 0},
+    {0xCB, (uint8_t []){0x5B}, 1, 0},
+    {0xCC, (uint8_t []){0x1B}, 1, 0},
+    {0xCD, (uint8_t []){0x1B}, 1, 0},
+    {0xCE, (uint8_t []){0x4F}, 1, 0},
+    {0xCF, (uint8_t []){0x24}, 1, 0},
+    {0xD0, (uint8_t []){0x2A}, 1, 0},
+    {0xD1, (uint8_t []){0x4E}, 1, 0},
+    {0xD2, (uint8_t []){0x5F}, 1, 0},
+    {0xD3, (uint8_t []){0x39}, 1, 0},
+    {ILI9881C_CMD_CNDBKxSEL, (uint8_t []){ILI9881C_CMD_BKxSEL_BYTE0, ILI9881C_CMD_BKxSEL_BYTE1, ILI9881C_CMD_BKxSEL_BYTE2_PAGE0}, 3, 0},
+    {0x35, (uint8_t []){0x00}, 1, 0},
+    {0x29, (uint8_t []){0x00}, 0, 0},
+
+    //============ Gamma END===========
+};
+
 static esp_err_t panel_ili9881c_del(esp_lcd_panel_t *panel)
 {
-    ili9881c_panel_t *ili9881c = __containerof(panel, ili9881c_panel_t, base);
+    ili9881c_panel_t *ili9881c = (ili9881c_panel_t *)panel->user_data;
 
     if (ili9881c->reset_gpio_num >= 0) {
         gpio_reset_pin(ili9881c->reset_gpio_num);
     }
+    // Delete MIPI DPI panel
+    ili9881c->del(panel);
     free(ili9881c);
-    return ESP_OK;
-}
-
-static esp_err_t panel_ili9881c_reset(esp_lcd_panel_t *panel)
-{
-    ili9881c_panel_t *ili9881c = __containerof(panel, ili9881c_panel_t, base);
-    esp_lcd_panel_io_handle_t io = ili9881c->io;
-
-    // perform hardware reset
-    if (ili9881c->reset_gpio_num >= 0) {
-        gpio_set_level(ili9881c->reset_gpio_num, ili9881c->reset_level);
-        vTaskDelay(pdMS_TO_TICKS(10));
-        gpio_set_level(ili9881c->reset_gpio_num, !ili9881c->reset_level);
-        vTaskDelay(pdMS_TO_TICKS(10));
-    } else { // perform software reset
-        ESP_RETURN_ON_ERROR(esp_lcd_panel_io_tx_param(io, LCD_CMD_SWRESET, NULL, 0), TAG, "send command failed");
-        vTaskDelay(pdMS_TO_TICKS(20)); // spec, wait at least 5ms before sending new command
-    }
+    ESP_LOGD(TAG, "del ili9881c panel @%p", ili9881c);
 
     return ESP_OK;
 }
 
 static esp_err_t panel_ili9881c_init(esp_lcd_panel_t *panel)
 {
-    ili9881c_panel_t *ili9881c = __containerof(panel, ili9881c_panel_t, base);
+    ili9881c_panel_t *ili9881c = (ili9881c_panel_t *)panel->user_data;
     esp_lcd_panel_io_handle_t io = ili9881c->io;
+    const ili9881c_lcd_init_cmd_t *init_cmds = NULL;
+    uint16_t init_cmds_size = 0;
+    uint8_t lane_command = ILI9881C_DSI_2_LANE;
+    bool is_command0_enable = false;
+    bool is_cmd_overwritten = false;
+
+    switch (ili9881c->lane_num) {
+    case 0:
+    case 2:
+        lane_command = ILI9881C_DSI_2_LANE;
+        break;
+    case 3:
+    case 4:
+        lane_command = ILI9881C_DSI_3_4_LANE;
+        break;
+    default:
+        ESP_LOGE(TAG, "Invalid lane number %d", ili9881c->lane_num);
+        return ESP_ERR_INVALID_ARG;
+    }
 
```

Comment:
question, should we always set the lane number first, then other parameters?

Reply:
In fact, the order of setting the lane number doesn't matter as long as it's set during initialization. However, sometimes different commands require page flipping, so handling the lane number first is more convenient.